src/detect-dataset.c

```diff
@@ -416,8 +489,25 @@ int DetectDatasetSetup (DetectEngineCtx *de_ctx, Signature *s, const char *rawst
     SCLogDebug("cmd %s, name %s",
         cmd_str, strlen(name) ? name : "(none)");
 
-    /* Okay so far so good, lets get this into a SigMatch
-     * and put it in the Signature. */
+    if (cmd == DETECT_DATASET_CMD_SET) {
+        // for set operation, we need one match, and one postmatch
```

Comment:
Q: Why can't it be just a postmatch?

Reply:
postmatch does not have a direct access to the content of the sticky buffer(s) (+ transforms)...

Does that answer your question ?

Reply:
> postmatch does not have a direct access to the content of the sticky buffer(s) (+ transforms)...

oh.

> Does that answer your question ?

yes. thank you very much! 🙇🏽‍♀️